lib/ExecutionControlEpic/Plugins/Gulp.js

```diff
@@ -57,16 +60,32 @@ export default {
       },
       controller: {
         onData(data: string, taskAPI: TaskAPI, helperAPI: HelperApi): void {
-          taskAPI.addDiagnostics([
-            {
-              type: "info",
-              message: {
-                text: helperAPI.outputToHTML(data.toString()),
-                html: true,
-              },
-              date: moment().unix(),
+          let dataFromCache = taskAPI.cache.get();
+          taskAPI.cache.clear();
+          let fullData = dataFromCache.reduce(
+            (datas: string, element: CacheBlob) => {
+              return datas + element.data;
             },
-          ]);
+            "",
+          );
+          fullData += data.toString();
+          let splitData = fullData.split("\n");
+          if (!fullData.endsWith("\n")) {
+            let dataToCache = splitData.pop();
+            taskAPI.cache.push(dataToCache);
+          }
+          splitData.forEach(line => {
+            return taskAPI.addDiagnostics([
+              {
+                type: "info",
+                message: {
+                  text: helperAPI.outputToHTML(line),
```

Comment:
You should import stripAnsi from "strip-ansi" and use it here : helperAPI.outputToHTML(stripAnsi(line))